app/projects/controller.js

```diff
@@ -11,21 +12,46 @@ export default Controller.extend({
   notify: service(),
   user: reads("session.data.user"),
   projects: reads("fetchProjectsOfUser.lastSuccessful.value"),
+  filteredProjects: reads("filterProjects.lastSuccessful.value"),
   tasks: reads("fetchTasksOfProject.lastSuccessful.value"),
+  customers: uniqBy("fetchCustomersOfProject.lastSuccessful.value", "id"),
+  loading: or(
+    "fetchTasksOfProject.isRunning",
+    "fetchCustomersOfProject.isRunning"
+  ),
 
   fetchProjectsOfUser: task(function*() {
     try {
+      let projects;
       if (this.get("user.isSuperuser")) {
-        return yield this.store.findAll("project");
+        projects = yield this.store.findAll("project");
+      } else {
+        projects = yield this.store.query("project", {
+          reviewer: this.get("user.id")
+        });
       }
-      return yield this.store.query("project", {
-        reviewer: this.get("user.id")
-      });
+
+      this.fetchCustomersOfProject.perform(projects);
```

Comment:
Is it on purpose that you don't await/yield here?

Reply:
Yes, IMO it's not necessary since the project selection will be disabled unless there is a selected customer anyways. Feel free to correct me though